src/core/ldap/domain/ldap-client.service.ts

```diff
@@ -248,38 +225,63 @@ export class LdapClientService {
     public async changeEmailAddressByPersonId(personId: PersonID, newEmailAddress: string): Promise<Result<PersonID>> {
         return this.mutex.runExclusive(async () => {
             this.logger.info('LDAP: changeEmailAddress');
+            const splitted: string[] = newEmailAddress.split('@');
+            if (!splitted || !splitted[1]) {
+                this.logger.error(`LDAP: Invalid email-address:${newEmailAddress}`);
+
+                return {
+                    ok: false,
+                    error: new LdapEmailAddressError(),
+                };
+            }
+            const domain: string = splitted[1];
+            const rootName: Result<string> = this.getRootNameOrError(domain);
+            if (!rootName.ok) return rootName;
             const client: Client = this.ldapClient.getClient();
             const bindResult: Result<boolean> = await this.bind();
             if (!bindResult.ok) return bindResult;
-
-            const searchResult: SearchResult = await client.search(`ou=oeffentlicheSchulen,dc=schule-sh,dc=de`, {
-                scope: 'sub',
-                filter: `(entryUUID=${personId})`,
-                attributes: ['mailPrimaryAddress', 'mailAlternativeAddress'],
-                returnAttributeValues: true,
-            });
+            const searchResult: SearchResult = await client.search(
+                `ou=${rootName.value},${LdapClientService.DC_SCHULE_SH_DC_DE}`,
+                {
+                    scope: 'sub',
+                    filter: `(entryUUID=${personId})`,
+                    attributes: [LdapClientService.MAIL_PRIMARY_ADDRESS, LdapClientService.MAIL_ALTERNATIVE_ADDRESS],
+                    returnAttributeValues: true,
+                },
+            );
             if (!searchResult.searchEntries[0]) {
+                this.logger.error(`Changing email-address in LDAP FAILED, no entry for personId:${personId}`);
                 return {
                     ok: false,
                     error: new LdapSearchError(LdapEntityType.LEHRER),
                 };
             }
 
-            const currentEmailAddressString: string | undefined = searchResult.searchEntries[0][
-                'mailPrimaryAddress'
+            // result can be a simple string or a string-array
+            let currentEmailAddressString: string | string[] | undefined = searchResult.searchEntries[0][
+                LdapClientService.MAIL_PRIMARY_ADDRESS
             ] as string;
+            if (Array.isArray(currentEmailAddressString)) {
+                currentEmailAddressString = (currentEmailAddressString as string[])[0];
+            }
```

Comment:
In diesem Fall könntest du auch einfach den Type 'unknown' für currentEmailAddress nutzen und dann anschließend mit runtime-typechecks den tatsächlichen Type prüfen und entsprechend behandeln. So kann die Type Assertion vermieden werden.